ldk/go/readme.md

```diff
@@ -11,9 +11,76 @@ go get -u github.com/open-olive/loop-development-kit/ldk/go
 Full documentation available on [pkg.go.dev](https://pkg.go.dev/github.com/open-olive/loop-development-kit/ldk/go).
 
 ### Loops
-This LDK can be used to write loops for Olive Helps. More detail about loops is available [here](docs/loops.md).
+Loops receive events and use them to generate relevant whispers. Loops choose which events they want to use and which they want to ignore.
 
-Example loops are available in the examples folder.
+This LDK can be used to write loops for Olive Helps. An example of the loop implementation is available [here](docs/loops.md).
+
+
+#### Interface
+
+Writing a loop boils down to writing an implementation for the `Loop` interface and serving it via `ldk.ServeLoopPlugin`.
+
+```go
+type Loop interface {
+	LoopStart(Sidekick) error
+	LoopStop() error
+}
+
+func main() {
+	...
+	ldk.ServeLoopPlugin(logger, loop) // ServeLoopPlugin, gives an ability to provide your own logger
+}	
+```
+
+**LoopStart** - The Loop should wait to start operating until this is called. The provided `Sidekick` reference should be stored in memory for continued use.
+
+**LoopStop** - The Loop should stop operating when this is called.
+
+#### Subscribing to Sensors
+
+Inside `LoopStart`, you can subscribe to various sensors. Here's an example of subscribing to a couple:
+
+```go
+func (c *Loop) LoopStart(sidekick ldk.Sidekick) error {
+	// ...
+
+	handler := func (text string, err error) {
+		// Respond to sensor event here...
+	}
+
+	if err := sidekick.Clipboard().Listen(l.ctx, handler); err != nil {
+		return err
+	}
+
+	if err := sidekick.UI().ListenSearchbar(l.ctx, handler); err != nil {
+		return err
+	}
+}
+```
+
+Loops do not need to emit whispers in a 1:1 relationship with events. Loops may not use events at all. Loops may only use some events. Loops may keep a history of events and only emit whispers when several conditions are met.
```

Comment:
Beginning each sentence here with "Loops" feels a little...samey? It just doesn't flow very well.

It doesn't have to be exactly this, but something like

"Loops do not need to emit whispers in a 1:1 relationship with events, and may not use events at all. They could only use some events, keep a history of events, or only emit whispers when several conditions are met."

Reply:
done